src/progress.js

```diff
@@ -1,4 +1,8 @@
 'use strict';
+// TODO needs to be able to have a width be determined
```

Comment:
why?  If you put it in a constrained width component, it should just take up the space given